### ISSUE TITLE
Why choose  "SoftmaxWithLoss"  for "loss_attr"

### ISSUE THREAD

**Issue by YuanEZhou** [peteanderson80/bottom-up-attention#43]
Hi @peteanderson80 , Thanks for your sharing.  
There are some objects has multiple attributes, why choose  "SoftmaxWithLoss"  for "loss_attr"?

**Comment by zhaohengz** [comment]
Hi, did anybody figure out the reason about the loss function?

**Comment by YuanEZhou** [comment]
Actually, the operation of this loss function is what we want.   “SoftmaxWithLoss”  was rewrited.   You can check the souce code at this link`https://github.com/peteanderson80/bottom-up-attention/blob/master/caffe/src/caffe/layers/softmax_loss_layer.cpp`

**Comment by zhaohengz** [comment]
Thanks! So what they implemented is something like `softmaxBinaryCrossEntorpy`. Did I get it right?

**Comment by YuanEZhou** [comment]
yeah,  this rewrited function try to maximize the sum of  all attribures' score .
